digiwf-integrations/digiwf-ticket-integration/digiwf-ticket-integration-core/src/main/java/de/muenchen/oss/digiwf/ticket/integration/adapter/out/s3/S3Adapter.java

```diff
@@ -0,0 +1,95 @@
+import de.muenchen.oss.digiwf.ticket.integration.application.port.out.LoadFilePort;
+import de.muenchen.oss.digiwf.ticket.integration.domain.model.FileContent;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.tika.Tika;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Slf4j
+@RequiredArgsConstructor
+public class S3Adapter implements LoadFilePort {
+
+    private final DocumentStorageFileRepository documentStorageFileRepository;
+    private final DocumentStorageFolderRepository documentStorageFolderRepository;
+    private final ProcessConfigApi processConfigApi;
+    private final List<String> supportedExtensions;
+
+    private final String APP_FILE_S3_SYNC_CONFIG = "app_file_s3_sync_config";
+
+    @Override
+    public List<FileContent> loadFiles(final List<String> filepaths, final String processDefinition) {
+        final String s3Storage = getDomainSpecificS3Storage(processDefinition).orElse(null);
+        final List<FileContent> contents = new ArrayList<>();
+        filepaths.forEach(path -> {
+            if (path.endsWith("/")) {
+                contents.addAll(getFilesFromFolder(path, s3Storage));
+            } else {
+                contents.add(getFile(path, s3Storage));
+            }
+        });
+        return contents;
+    }
+
+    private List<FileContent> getFilesFromFolder(String folderpath, final String domainSpecificS3Storage) {
+        try {
+            final List<FileContent> contents = new ArrayList<>();
+            final Set<String> filepath = documentStorageFolderRepository.getAllFilesInFolderRecursively(folderpath, domainSpecificS3Storage).block();
+            filepath.forEach(file -> contents.add(getFile(file, domainSpecificS3Storage)));
+            return contents;
+        } catch (final DocumentStorageException | DocumentStorageServerErrorException |
+                       DocumentStorageClientErrorException e) {
+            throw new BpmnError("LOAD_FOLDER_FAILED", "An folder could not be loaded from url: " + folderpath);
+        }
+    }
+
+    private FileContent getFile(String filepath, final String domainSpecificS3Storage) {
+        try {
+            final Tika tika = new Tika();
+            final byte[] bytes = this.documentStorageFileRepository.getFile(filepath, 3, domainSpecificS3Storage);
+            final String mimeType = tika.detect(bytes);
+            final String filename = FilenameUtils.getName(filepath);
+
+            // check if mimeType exists
+            supportedExtensions
+                    .stream()
+                    .filter(extension -> extension.equals(mimeType))
+                    .findAny()
+                    .orElseThrow(() -> new BpmnError("FILE_TYPE_NOT_SUPPORTED", "The type of this file is not supported: " + filepath));
+
+            return new FileContent(mimeType, filename, bytes);
```

Comment:
Wieso hast du den die ganze Filetype-Verification nicht gleich in das FileRepository bzw. die S3-Client-Lib gepackt?

Reply:
Ich habe mir den S3 Adapter zum großteil von der DMS Integration abgeschaut. Eigentlich könnte man den ganzen S3 Adapter in eine extra Lib bzw. die S3-Client-Lib auslagern. Dafür habe ich ein Ticket aufgemacht https://github.com/it-at-m/digiwf-core/issues/1442